backend/api/llm_providers.py

```
@@ -0,0 +1,188 @@
+    return "***"
+
+async def check_admin_access(user_id: str = Depends(get_current_user)):
+    # Mocking role claim check for now since auth.py just returns string
+    user_roles = ["admin"] if user_id == "admin" else []
+    if "admin" not in user_roles:
+        raise HTTPException(status_code=403, detail="Admin access required")
+    return user_id
+
+@router.get("", response_model=List[LLMProviderResponse])
+async def list_providers(
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)